src/context/global.jsx

```diff
@@ -17,11 +18,28 @@ export const GlobalContextProvider = ({ children }) => {
     const getUser = async () => {
       const token = getCookie('token');
 
-      if (token) {
-        const data = await httpGet(`me`, token);
+      let data;
+      if (token !== null) {
+        data = await httpGet('test/user', token);
+      }
+
+      if (token && !data.message) {
+        setUser((prevState) => ({
+          ...prevState,
+          userInfo: data,
+          isLogin: true
+        }));
+      } else if (token && data.message) {
+        const refreshToken = getCookie('refreshToken');
+        const newAccessToken = await httpPost('auth/refreshtoken', {
+          refreshToken: `${refreshToken}`
+        });
+        await setCookie('token', newAccessToken.accessToken, 1);
+        const newToken = getCookie('token');
+        const newData = await httpGet('test/user', newToken);
         setUser((prevState) => ({
           ...prevState,
-          userInfo: data?.user,
+          userInfo: newData,
           isLogin: true
```

Comment:
use setUser once. like
`
let data;

if (blabla === blabla2) {
  data = {}
} else if (blabla === blabla3) {
  data = {}
}

setUser((prevState) => ({
          ...prevState,
          userInfo: data,
          isLogin: true
        }));

`

Reply:
`(async () => {
      const token = getCookie('token');

      let data;
      if (token !== null) {
        data = await httpGet('user', token);
      }

      let userInfo;
      if (token && !data.message) {
        userInfo = data;
      } else if (token && data.message) {
        const refreshToken = getCookie('refreshToken');
        const newAccessToken = await httpPost('auth/refreshtoken', {
          refreshToken: `${refreshToken}`
        });
        await setCookie('token', newAccessToken.accessToken, 1);
        const newToken = getCookie('token');
        userInfo = await httpGet('user', newToken);
      }

      if (userInfo) {
        setUser((prevState) => ({
          ...prevState,
          userInfo,
          isLogin: true
        }));
      }
    })()`

Reply:
no needed to have getUser(), use IIFE